### PR TITLE
Fix NumberParsing for double with exponent

### PR DESCRIPTION
**Pull request by StefH**

### CODE REVIEW

File: src/System.Linq.Dynamic.Core/Parser/NumberParser.cs [hunk 1]
```diff
@@ -29,73 +44,73 @@ public object ParseNumber(string text, Type type)
...
                 }
                 if (tp == typeof(float))
                 {
-                    return float.Parse(text, _config.NumberParseCulture);
+                    return float.Parse(text, _culture);
                 }
                 if (tp == typeof(double))
                 {
-                    return double.Parse(text, _config.NumberParseCulture);
+                    return double.Parse(text, _culture);
```
Comment (alexweav) [inline src/System.Linq.Dynamic.Core/Parser/NumberParser.cs:109]: Nice, I think the default will work great for us. FWIW, the default value seems to be `NumberStyles.Float | NumberStyles.AllowThousands`. According to the [somewhat confusing NumberStyles docs](https://docs.microsoft.com/en-us/dotnet/api/system.globalization.numberstyles?view=net-5.0#system-globalization-numberstyles-allowtrailingsign), this actually does exclude one flag that was enabled prior to this change, which was `NumberStyles.AllowTrailingSign`.

This flag is really strange, it lets me specify the sign after the number like in [this fiddle](https://dotnetfiddle.net/GbPFqn). I haven't seen any actual use cases of this in the wild, and I don't know if the parser even supports it. I definitely don't need trailing signs, just wanting to make sure that we aren't regressing this edge case.

File: test/System.Linq.Dynamic.Core.Tests/DynamicExpressionParserTests.cs [hunk 2]
```diff
@@ -644,21 +644,114 @@ public void DynamicExpressionParser_ParseLambda_ParameterExpressionMethodCall_Re
...
+                new object[] { "de-DE", "-42d", -42d },
+                new object[] { "de-DE", "3,215d", 3.215d },
+                new object[] { "de-DE", "1,2345E-4", 0.00012345d },
+                new object[] { "de-DE", "1,2345E4", 12345d },
+
+                new object[] { null, "1D", 1d },
+                new object[] { null, "1d", 1d },
+                new object[] { null, "-42d", -42d },
+                new object[] { null, "3.215d", 3.215d },
+                new object[] { null, "1.2345E-4", 0.00012345d },
```
Comment (alexweav) [inline test/System.Linq.Dynamic.Core.Tests/DynamicExpressionParserTests.cs:732]: These are looking great. We saw a few cases where there was a lowercase `e` in the exponent. `double.Parse` handles this with no issue, but it may be worth including a test for it.

File: src/System.Linq.Dynamic.Core/Parser/NumberParser.cs (hunk 1 above)
Comment (alexweav) [reply src/System.Linq.Dynamic.Core/Parser/NumberParser.cs:109]: I just [tried it](https://dotnetfiddle.net/HHbzpm), and it looks like trailing signs are already rejected elsewhere in the parser. So, this isn't a regression at all, we can ignore the trailing sign flag 👍

File: test/System.Linq.Dynamic.Core.Tests/DynamicExpressionParserTests.cs (hunk 2 above)
Comment (StefH) [reply test/System.Linq.Dynamic.Core.Tests/DynamicExpressionParserTests.cs:732]: I did add this test.

File: src/System.Linq.Dynamic.Core/Parser/NumberParser.cs (hunk 1 above)
Comment (StefH) [reply src/System.Linq.Dynamic.Core/Parser/NumberParser.cs:109]: OK

**Review by alexweav — APPROVED** [review]
Thanks again!

**Comment by StefH** [comment]
https://github.com/zzzprojects/System.Linq.Dynamic.Core/issues/531
